Hints to overload protection and ANALYZE command (#325)
crate/crate-howtos: docs/performance/inserts/tuning.rst

insert performance, but it also worsens durability. If a node crashes before a
translog has been synced, those opperations will be lost.

Overload Protection
-------------------

> The `Overload Protection`_ settings control how many resources operations like 

Reviewer: That may also work, using Intersphinx references.
```suggestion
The :ref:`Overload Protection <crate-reference:overload_protection>` settings control how many resources operations like 
```

> `INSERT INTO FROM QUERY` or `COPY` can use.

Reviewer: .rst uses pairs of "`".
```suggestion
``INSERT INTO FROM QUERY`` or ``COPY`` can use.
```

The default values serve as a starting point for an algorithm that dynamically 
adapts the effective concurrency limit based on the round-trip time of requests. 
Whenever one of these settings is updated, the previously calculated effective 
concurrency is reset.

Please update the settings accordingly, especially if you are benchmarking insert
performance.

> 

Reviewer: A few trailing spaces sneaked in here. Can I humbly ask you to configure your editor to strip those, or take extra care when committing?

![image](https://github.com/crate/crate-howtos/assets/453543/d6d131ea-91ee-4050-b500-19906ae99ebe)

interval, you can expect to see performance improvements if you increase this
value.

Calculating statistics
----------------------

> After loading larger amounts of data into new or exiting tables, it is recommended

Reviewer: ```suggestion
After loading larger amounts of data into new or existing tables, it is recommended
```

> to (re-)calculate the statistics by executing the ``ANALYZE`` command. 

Reviewer: ```suggestion
to re-calculate the statistics by executing the ``ANALYZE`` command. 
```

The statistics will be used by the query optimizer to generate better execution plans. 

> This calculation of statistics is performed periodically and the maximum amount of bytes 
> per second that can be read from data nodes for the collection of statistics are throttled.

Reviewer: ```suggestion
The calculation of statistics happens periodically. The bandwidth used for collecting statistics
is limited by applying throttling based on the maximum amount of bytes per second that can
be read from data nodes.
```

Please refer to the `ANALYZE`_ documentation for further information how to change the

> calculation interval and how to configure the throttling. 

Reviewer: ```suggestion
calculation interval, and how to configure throttling settings.
```

.. _fulltext indexes: https://crate.io/docs/crate/reference/en/latest/sql/fulltext.html
.. _natural primary key: https://en.wikipedia.org/wiki/Natural_key
.. _OPTIMIZE: https://crate.io/docs/crate/reference/en/latest/sql/reference/optimize.html
.. _ANALYZE: https://cratedb.com/docs/crate/reference/en/latest/sql/statements/analyze.html
.. _refresh_interval: https://crate.io/docs/crate/reference/en/latest/sql/reference/create_table.html#refresh-interval
.. _Solid-State Drives: https://en.wikipedia.org/wiki/Solid-state_drive
.. _surrogate primary key: https://en.wikipedia.org/wiki/Surrogate_key
.. _system column: https://crate.io/docs/crate/reference/en/latest/sql/administration/system_columns.html
.. _translog.durability: https://crate.io/docs/crate/reference/en/latest/sql/reference/create_table.html#translog-durability
.. _turning column indexes off: https://crate.io/docs/crate/reference/en/latest/sql/ddl/indices_full_search.html#disable-indexing

> :: _Overload Protection: https://cratedb.com/docs/crate/reference/en/latest/config/cluster.html#overload-protection

Reviewer: Maybe that is a problem?

```
/home/runner/work/crate-howtos/crate-howtos/docs/performance/inserts/tuning.rst:169: WARNING: Explicit markup ends without a blank line; unexpected unindent.
```
-- https://github.com/crate/crate-howtos/actions/runs/6874179754/job/18695387749?pr=325#step:4:412

> :: _Overload Protection: https://cratedb.com/docs/crate/reference/en/latest/config/cluster.html#overload-protection

Reviewer: There is a better way to link to other Spinx documentation, like
```
:ref:`crate-reference:overload_protection`
```

Reviewer: See https://github.com/crate/cratedb-guides/pull/6#pullrequestreview-1734102915:

> :: _Overload Protection: https://cratedb.com/docs/crate/reference/en/latest/config/cluster.html#overload-protection

Reviewer: There is an error here.

```suggestion
.. _Overload Protection: https://cratedb.com/docs/crate/reference/en/latest/config/cluster.html#overload-protection
```
